mock_app/api/routes/schedule.py

```
@@ -0,0 +1,112 @@
+    await ctx.task.progress(FULL_PROGRESS)
+    ctx.logger.info("Schedule ok: task %s completed", ctx.meta.task_id)
+
+
+@schedule_router.task(
+    "/use-cases/cancel",
+    id="mock.use-cases.cancel",
+    name="schedule-use-case-cancel",
+    description="Fail the task by raising CancelError",
+    cron="*/3 * * * *",
+)
+async def handle_cancel(ctx: ScheduleContext) -> None:  # ruff:ignore[unused-async] - schedule handlers are async by contract
```

Ruff (0.16.1)

[warning] Invalid rule code in suppression: unused-async

Enable `lint.preview` to use rule names

(RUF102)